lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -502,6 +502,38 @@ static LogicalResult verifyInstanceOp(InstanceOp &instance) {
     return failure();
   }
 
+  // Check that the result type is consistent with its module.
+  if (auto referencedFModule = dyn_cast<FModuleOp>(referencedModule)) {
+    auto bundle = instance.getResult()
+                      .getType()
+                      .cast<FIRRTLType>()
+                      .getPassiveType()
+                      .cast<BundleType>();
+    auto bundleElements = bundle.getElements();
+    size_t e = bundleElements.size();
+
+    if (e != referencedFModule.getNumArguments()) {
+      instance.emitOpError()
+          << "has a wrong size of bundle type, expected size is "
+          << referencedFModule.getNumArguments() << " but got " << e;
```

Comment:
I think it would be useful to use a note to point at the `FModuleOp` definition. I think it's something like:
 
```suggestion
      auto diag = instance.emitOpError()
          << "has a wrong size of bundle type, expected size is "
          << referencedFModule.getNumArguments() << " but got " << e;
      diag.attachNote(referencedFModule.getLoc()) << "original module declared here";
```

Reply:
Thanks!